packages/@ourworldindata/grapher/src/controls/Controls.tsx

```diff
@@ -1,18 +1,500 @@
+                selection instanceof SelectionArray
+                    ? selection.selectedEntityNames
+                    : Array.isArray(selection)
+                    ? selection
+                    : []
+
+        return !hideZoomToggle && type === ScatterPlot && entities.length > 0
+    }
+
+    @computed get showNoDataAreaToggle(): boolean {
+        return (
+            !this.manager.hideNoDataAreaToggle &&
+            this.manager.type === Marimekko &&
+            this.manager.xColumnSlug !== undefined
+        )
+    }
+
+    @computed get showAbsRelToggle(): boolean {
+        const { type, canToggleRelativeMode, hasTimeline, xOverrideTime } =
+            this.manager
+        if (!canToggleRelativeMode) return false
+        if (type === ScatterPlot)
+            return xOverrideTime === undefined && !!hasTimeline
+        return [
+            StackedArea,
+            StackedBar,
+            StackedDiscreteBar,
+            ScatterPlot,
+            LineChart,
+            Marimekko,
+        ].includes(type)
+    }
+
+    @computed get showFacetControl(): boolean {
+        const { filledDimensions, availableFacetStrategies, type } =
+            this.manager
+
+        // if there's no choice to be made, don't display a lone button
+        if (availableFacetStrategies.length <= 1) return false
+
+        // heuristic: if the chart doesn't make sense unfaceted, then it probably
+        // also makes sense to let the user switch between entity/metric facets
+        if (!availableFacetStrategies.includes(FacetStrategy.none)) return true
+
+        const showFacetControlChartType = [
+            StackedArea,
+            StackedBar,
+            StackedDiscreteBar,
+            LineChart,
+        ].includes(type)
+
+        const hasProjection = filledDimensions.some(
+            (dim) => dim.display.isProjection
+        )
+
+        return showFacetControlChartType && !hasProjection
+    }
+
+    @computed get showSettingsMenuToggle(): boolean {
+        return !!(
+            this.showYScaleToggle ||
+            this.showXScaleToggle ||
+            this.showFacetYDomainToggle ||
+            this.showZoomToggle ||
+            this.showNoDataAreaToggle ||
+            this.showFacetControl ||
+            this.showAbsRelToggle
+        )
+
+        // TODO: add a showCompareEndPointsOnlyTogggle to complement compareEndPointsOnly
+    }
+
+    @action.bound toggleVisibility(e: React.MouseEvent): void {
+        this.visible = !this.visible
+        if (this.visible) this.shouldRender = true
+        this.drawer?.classList.toggle("active", this.visible)
+        e.stopPropagation()
+    }
+
+    @action.bound onAnimationEnd(): void {
+        if (!this.visible) this.shouldRender = false
+    }
+
+    @computed get manager(): SettingsMenuManager {
+        return this.props.manager
+    }
+
+    @computed get chartType(): string {
+        const { type } = this.manager
+        return type.replace(/([A-Z])/g, " $1")
```

Comment:
Out of interest, why is this necessary? Shouldn't the `type` be one of [ChartTypeName](https://github.com/owid/owid-grapher/blob/a3d32f96431cbc43b74d8d00b9539432c52044d5/packages/%40ourworldindata/grapher/src/core/GrapherConstants.ts#L5)?